app/src/sharedTest/java/org/oppia/app/topic/train/TopicTrainFragmentTest.kt

```diff
@@ -2,36 +2,138 @@ package org.oppia.app.topic.train
 import org.oppia.app.topic.TopicActivity
 import org.oppia.util.threading.BackgroundDispatcher
 import org.oppia.util.threading.BlockingDispatcher
 import javax.inject.Singleton
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.matcher.ViewMatchers.isChecked
+import org.oppia.app.R
+import org.oppia.app.RecyclerViewMatcher
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import androidx.test.rule.ActivityTestRule
 
 /** Tests for [TopicTrainFragment]. */
 @RunWith(AndroidJUnit4::class)
 class TopicTrainFragmentTest {
 
+  private var skillIdListIntent = ArrayList<String>()
+
+  @get:Rule
+  var activityTestRule: ActivityTestRule<TopicActivity> = ActivityTestRule(
+    TopicActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false
+  )
+
+  @Before
+  fun setUp() {
+    Intents.init()
+    skillIdListIntent.add("test_skill_id_0")
+    skillIdListIntent.add("test_skill_id_1")
+  }
+
+  @Test
+  fun testTopicTrainFragment_loadFragment_displaySkills_startButtonIsInactive() {
+    ActivityScenario.launch(TopicActivity::class.java).use {
+      onView(withId(R.id.master_skills_text_view)).check(matches(withText(R.string.topic_train_master_these_skills)))
+      onView(withRecyclerView(R.id.skill_recycler_view).atPosition(0))
+        .check(matches(hasDescendant(withId(R.id.skill_check_box))))
+      onView(withId(R.id.topic_train_start_button)).check(matches(not(isClickable())))
+    }
+  }
+
   @Test
-  fun testTopicTrainFragment_loadFragment_textIsDisplayed() {
+  fun testTopicTrainFragment_loadFragment_selectSkills_deselectSkills_startButtonIsInactive() {
     ActivityScenario.launch(TopicActivity::class.java).use {
-      onView(withId(R.id.dummy_text_view)).check(matches(withText("This is dummy TextView for testing")))
+      onView(withId(R.id.master_skills_text_view)).check(matches(withText(R.string.topic_train_master_these_skills)))
+      onView(withRecyclerView(R.id.skill_recycler_view).atPosition(0)).perform(click())
+      onView(withId(R.id.topic_train_start_button)).check(matches(isClickable()))
+      onView(withRecyclerView(R.id.skill_recycler_view).atPosition(0)).perform(click())
+      onView(withId(R.id.topic_train_start_button)).check(matches(not(isClickable())))
     }
   }
 
+  @Test
+  fun testTopicTrainFragment_loadFragment_selectSkills_clickStartButton_skillListTransferSuccessfully() {
+    activityTestRule.launchActivity(null)
+    onView(withId(R.id.master_skills_text_view)).check(matches(withText(R.string.topic_train_master_these_skills)))
+    onView(withRecyclerView(R.id.skill_recycler_view).atPosition(0)).perform(click())
+    onView(withRecyclerView(R.id.skill_recycler_view).atPosition(1)).perform(click())
+    onView(withId(R.id.topic_train_start_button)).check(matches(isClickable()))
+    onView(withId(R.id.topic_train_start_button)).perform(click())
+    intended(hasComponent("org.oppia.app.topic.questionplayer.QuestionPlayerActivity"))
```

Comment:
Could we depend directly on QuestionPlayerActivity's class & to retrieve the intent extra KEY? Otherwise refactoring may be tricky, or renaming the intent key could break this test.

Reply:
Resolved. Used QuestionPlayerActvity name and key directly from the QuestionPlayerActvity file.